controllers/lvmcluster_controller.go

```diff
@@ -309,12 +322,15 @@ func (r *LVMClusterReconciler) getRunningPodImage(ctx context.Context) error {
 		// 'POD_NAME' and 'POD_NAMESPACE' are set in env of lvm-operator when running as a container
 		podName := os.Getenv("POD_NAME")
 		if podName == "" {
-			return fmt.Errorf("failed to get pod name env variable")
+			err := fmt.Errorf("failed to get pod name env variable")
+			r.Log.Error(err, "POD_NAME env variable is not set.")
+			return err
 		}
 
 		pod := &corev1.Pod{}
 		if err := r.Get(ctx, types.NamespacedName{Name: podName, Namespace: r.Namespace}, pod); err != nil {
-			return fmt.Errorf("failed to get pod %s in namespace %s", podName, r.Namespace)
+			r.Log.Error(err, "Failed to get pod.")
```

Comment:
Will this include information about the pod that was requested?

Reply:
Yes error will have it, This is the error which gets generated `pods "test" not found` when I ran the below code.

```
pod := &corev1.Pod{}                                    
err = fakeReconciler.Client.Get(context.TODO(), types.NamespacedName{Name: "test", Namespace: "test"}, pod)                                                       
fmt.Println(err)
```